pxr/imaging/hdSt/materialXFilter.cpp

```diff
@@ -61,31 +61,86 @@ TF_DEFINE_PRIVATE_TOKENS(
     ((defaultInput, "default"))
 
     // Opacity Parameters
+    (UsdPreviewSurface)
     (opacity)
     (opacityThreshold)
+
+    (standard_surface)
     (transmission)
 
+    (open_pbr_surface)
+    (transmission_weight)
+    (geometry_opacity)
+
+    (gltf_pbr)
+    (alpha_mode)
+    (alpha_cutoff)
+    (alpha)
+
+    (convert)
+    (ND_convert_color4_surfaceshader)
+    (ND_convert_vector4_surfaceshader)
+
     // Fallback Dome Light Tokens
     (domeLightFallback)
     (ND_image_color3)
     (file)
 
     // Colorspace Tokens
-    (colorSpace)
     (sourceColorSpace)
+
+    // Anonymization constants
+    (NG_Anonymized)
+
+    // Primvar detection constants
+    (geompropvalue)
 );
 
+TF_DEFINE_PRIVATE_TOKENS(
+    _topologicalTokens,
+    // This represents living knowledge of the internals of the MaterialX shader generator
+    // for both GLSL and Metal. Such knowledge should reside inside the generator class
+    // provided by MaterialX.
```

Comment:
To be submitted to MaterialX for 1.39.

Reply:
Is the intent here to have bool MaterialXIsNodeTopological(std::string name) or something?

Reply:
We'd definitely be interested in this code moving into MX, for the record :D.

Reply:
Yes, the information being dependent on ShaderGen target, it should probably be `static bool MaterialX::GlslShaderGenerator::isNodeDefTopological(MaterialX::NodeDefPtr const&)`

I do love the fact that we can get all the required info from Sdr and compare via TfTokens without having to use the MaterialX library, but in this case correctness trumps convenience, so this will definitely be added to MaterialX. If performance matters, we can always cache the result since the return value will always be the same for a given nodeId.